Copilot-instructions: Add new sections per Copilot issue & review tests

wadepickett opened the pull request:
Contributes to:
#35991
#36113

Add new sections collaborated with @tdykstra based on many copilot issues and review handling tests.
Reformat to checkboxes so Copilot stops skipping instructions per current best practice.
Break the original instruction into two documents with a new doc that centers just on code instructions.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [.github/copilot-code-instructions.md](https://github.com/dotnet/AspNetCore.Docs/blob/4bb0234b3ac1e6a79cb1635ec8d95457ebc45984/.github/copilot-code-instructions.md) | [Copilot Code Instructions for `dotnet/AspNetCore.Docs`](https://review.learn.microsoft.com/en-us/aspnet/core/.github/copilot-code-instructions?branch=pr-en-us-36136) |
| [.github/copilot-instructions.md](https://github.com/dotnet/AspNetCore.Docs/blob/4bb0234b3ac1e6a79cb1635ec8d95457ebc45984/.github/copilot-instructions.md) | [.github/copilot-instructions](https://review.learn.microsoft.com/en-us/aspnet/core/.github/copilot-instructions?branch=pr-en-us-36136) |


<!-- PREVIEW-TABLE-END -->

tdykstra commented:
*f you want to wait until I am done drop them in.**
When you drop them in Please alert me via message on Teams

wadepickett commented:
...added code snippet folder instructions.

wadepickett commented:
@tdykstra,  I added all new sections and instruction lines I know of to reflect our testing and resulting instruction we added that proved to work properly.

If I missed any, please do add more.

After it is all in, we can consider if we need to break the document into 2 or three sub documents or leave it as is.

For the future, I think there is a need for specifying creating proper .http test files for all API app examples it adds in the future, such as for Minimal API and for Controller based API.  But I'll leave that for another day.

wadepickett commented:
Copilot found extra spaces, fixed it. Ready for review
